unreal/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+//! API to process Unreal Engine 4 crashes
+#![warn(missing_docs)]
+
+extern crate byteorder;
+extern crate bytes;
+extern crate compress;
+extern crate failure;
+
+use std::io::{self, Cursor, Read};
+
+use bytes::{Buf, Bytes};
+use compress::zlib;
+use failure::Fail;
+
+struct Header {
+    pub directory_name: String,
+    pub file_name: String,
+    pub uncompressed_size: i32,
+    pub file_count: i32,
+}
+
+/// Meta-data about a file within a UE4 crash file
+#[derive(Debug)]
+pub struct CrashFileMeta {
+    /// The original index within the UE4 crash file
+    pub index: usize,
+    /// File name
+    pub file_name: String,
+    /// Start of the file within crash dumb
+    pub offset: usize,
+    /// Length of bytes from offset
+    pub len: usize,
+}
+
+/// Errors related to parsing an UE4 crash file
+#[derive(Fail, Debug)]
+pub enum Unreal4ParseError {
```

Comment:
_nit_: Naming guidelines would call this `ParseUnreal4Error`, but you could also just call it `Unreal4Error`.